diff -Nur a/Pods/MKStoreKit/MKStoreKit.m b/Pods/MKStoreKit/MKStoreKit.m
--- a/Pods/MKStoreKit/MKStoreKit.m	2015-01-10 21:11:49.000000000 +0100
+++ b/Pods/MKStoreKit/MKStoreKit.m	2015-01-10 22:12:09.000000000 +0100
@@ -122,9 +122,8 @@
 
 -(NSString*) purchaseRecordFilePath {
   
-  NSString *documentDirectory = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory,
-                                                                    NSUserDomainMask, YES).firstObject;
-  return [documentDirectory stringByAppendingPathComponent:@"purchaserecord.plist"];
+  NSURL *containerURL = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.com.sylver.NAStify"];
+  return [containerURL.path stringByAppendingPathComponent:@"purchaserecord.plist"];
 }
 
 -(void) restorePurchaseRecord {
@@ -405,7 +404,7 @@
         
       case SKPaymentTransactionStateFailed:
         NSLog(@"Transaction Failed with error: %@", transaction.error);
-        [queue finishTransaction:transaction];
+        [self failedTransaction:transaction fromQueue:queue];
         break;
         
       case SKPaymentTransactionStatePurchased:
@@ -443,8 +442,13 @@
 
 - (void) failedTransaction: (SKPaymentTransaction *)transaction {
   
+  [self failedTransaction:transaction fromQueue:[SKPaymentQueue defaultQueue]];
+}
+
+- (void) failedTransaction:(SKPaymentTransaction *)transaction fromQueue:(SKPaymentQueue *)queue {
+
   NSLog(@"Transaction Failed with error: %@", transaction.error);
-  [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
+  [queue finishTransaction:transaction];
   [[NSNotificationCenter defaultCenter] postNotificationName:kMKStoreKitProductPurchaseFailedNotification
                                                       object:transaction.payment.productIdentifier];
 }
