diff -Nur a/lib/include/curl/curlbuild.h b/lib/include/curl/curlbuild.h
--- a/lib/include/curl/curlbuild.h	2014-06-06 02:06:18.000000000 +0200
+++ b/lib/include/curl/curlbuild.h	2014-06-06 02:01:39.000000000 +0200
@@ -160,7 +160,11 @@
 #endif
 
 /* The size of `long', as computed by sizeof. */
+#ifdef __LP64__
+#define CURL_SIZEOF_LONG 8
+#else
 #define CURL_SIZEOF_LONG 4
+#endif
 
 /* Integral data type used for curl_socklen_t. */
 #define CURL_TYPEOF_CURL_SOCKLEN_T socklen_t
